README.md

```diff
@@ -154,6 +154,119 @@ Coming soon: guidelines, specs and code documentation. Check out the examples fo
 
 See [the examples and demos](./demo/) for more examples.
 
+## Usage in Swift
+
+To use the library in Swift, it first must be compiled using the instructions above, then the library and header must be deployed manually to /usr:
+
+```bash
+sudo cp testcontainers-c/testcontainers-c.h /usr/include
+sudo cp testcontainers-c/testcontainers-c.so /usr/lib/libtestcontainers-c.so
+```
+
+This is required so that Swift can actually find the header and library. On some systems, it may be required to deploy the *.so file to `/usr/lib64` instead. Try this if Swift is unable to find it when deployed to `/usr/lib` as shown above.
+
+Next, create a new directory in your Swift project `Sources/` directory called `CTestContainers` and add a file called `CTestContainers.h` with the following content:
+
+```c
+#include <testcontainers-c.h>
+```
+
+Then, create a file called `module.modulemap` in the same directory with this:
+
+```c
+module CTestContainers [system] {
+    header "CTestContainers.h"
+    link "testcontainers-c"
+    export *
+}
+```
+
+Then, add a the following definitions to the `targets` in `Package.swift`:
+
+```swift
+.systemLibrary(name: "CTestContainers")
+```
+
+It can be then included as a dependency for a target, like this:
+
+```swift
+.executableTarget(
+    name: "my-swift-app",
+    dependencies: [
+        "CTestContainers"
+    ]
+),
+```
+
+This module can then be imported and used from inside the application. However, the only challenge is dealing with C-style strings that are typically utf8 and are null terminated, so they require a bit of extra support for translation in Swift.
+
+Here is the `generic-containers` demo translated into Swift, with an exstension to properly translate a `String` to a utf8 string for C:
```

Comment:
I guess it will get easier if we have a C++ binding header with `std::string`

Reply:
I agree with that. Also, Swift has some decent C++ interop these days, so it would make things much smoother to have a C++ interface with std::string being used.